core/src/test/java/com/linecorp/armeria/server/ServerTlsValidationTest.java

```diff
@@ -0,0 +1,84 @@
+
+import java.security.KeyStore;
+
+import javax.net.ssl.KeyManagerFactory;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+class ServerTlsValidationTest {
+
+    // Note: When key store password is not given to key store, it is expected that an exception occurs when
+    //       Server.builder().build() method is called. But when users use JKS key store, the exception is never
+    //       raised. (In case of PKCS12 key store, the exception is raised as expected.) Not sure why this
+    //       happens and this test needs to be updated when investigation completes.
+    @Disabled
+    @Test
+    void testJksKeyStoreWithNullPassword() {
+        /*
+         * Dummy keystore generation
+         * keytool -genkeypair -keyalg RSA -keysize 2048 -storetype JKS -keystore keystore.jks -validity 3650
+         * key store password = password
+         * key password = keypassword
+         */
+        assertThatThrownBy(() -> {
+            final KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(getClass().getResource("keystore.jks").openStream(), null);
+
+            final KeyManagerFactory kmf = KeyManagerFactory.getInstance(
+                    KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, "keypassword".toCharArray());
+
+            final Server server = Server.builder()
+                                        .service("/", (ctx, res) -> HttpResponse.of(HttpStatus.OK))
+                                        .tls(kmf, sslContextBuilder -> {})
+                                        .build();
+        }).isInstanceOf(RuntimeException.class)
+          .hasMessageContaining("failed to validate SSL/TLS configuration");
+    }
+
+    @Test
+    void testPkcs12KeyStoreWithNullPassword() throws Exception {
+        /*
+         * Dummy keystore generation
+         * keytool -genkeypair -keyalg RSA -keysize 2048 -storetype PKCS12 -keystore keystore.p12 -validity 3650
+         * keypassword = keystorepassword = password
+         */
+        assertThatThrownBy(() -> {
+            final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+            keyStore.load(getClass().getResource("keystore.p12").openStream(), null);
+
+            final KeyManagerFactory kmf =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, "password".toCharArray());
+
+            final Server server = Server.builder()
+                                        .service("/", (ctx, res) -> HttpResponse.of(HttpStatus.OK))
+                                        .tls(kmf, sslContextBuilder -> {})
+                                        .build();
+        }).isInstanceOf(RuntimeException.class)
+          .hasMessageContaining("failed to validate SSL/TLS configuration");
```

Comment:
Could you reduce `assertThatThrownBy` coverage? To catch error in exact place. 😄

Reply:
Fixed it. Thank you.